src/lighhouse.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include <glm/gtx/norm.hpp>
+#include <kvf/time.hpp>
+#include <le2d/context.hpp>
+#include <le2d/drawable/shape.hpp>
+#include <le2d/event.hpp>
+#include <le2d/renderer.hpp>
+#include <le2d/service_locator.hpp>
+#include "glm/ext/vector_float2.hpp"
+#include "gsl/pointers"
+#include "le2d/texture.hpp"
+namespace miracle {
+class Lighthouse {
+  public:
+	explicit Lighthouse(gsl::not_null<le::ServiceLocator const*> services);
+	void rotate_towards_cursor(glm::vec2 const& cursor_pos);
+	void render(le::Renderer& renderer) const;
+
+  private:
+	le::drawable::Circle m_sprite{};
+	std::optional<le::Texture> m_texture;
+	gsl::not_null<le::ServiceLocator const*> m_services;
```

Comment:
Order of members is important in C++. Theoretically `m_services` should be the first member since it's an injected dependency. In this case it's not that relevant, it's just a plain pointer. But say it was `le::Context`, that would get destroyed before `m_texture`, which would be very bad.